packages/oats-runtime/src/redirect.ts

```diff
@@ -0,0 +1,72 @@
+import type { Response } from './server';
+import encodeUrl = require('encodeurl');
+import escapeHtml = require('escape-html');
+
+const REDIRECT_STATUSES = [300, 301, 302, 303, 305, 307, 308] as const;
+const DEFAULT_REDIRECT_STATUS = 302;
+const TEXT_HTML_CONTENT_TYPE = 'text/html';
+const TEXT_PLAIN_CONTENT_TYPE = 'text/plain';
+
+export type RedirectStatus = typeof REDIRECT_STATUSES[number];
+
+type DefaultValue<ContentType extends string> = ContentType extends
+  | typeof TEXT_HTML_CONTENT_TYPE
+  | typeof TEXT_PLAIN_CONTENT_TYPE
+  | `${typeof TEXT_HTML_CONTENT_TYPE};${string}`
+  | `${typeof TEXT_PLAIN_CONTENT_TYPE};${string}`
+  ? string
+  : null;
+
+export function redirect<
```

Comment:
I'm kinda wondering why we need all of this code when plain helpers around runtime.json and runtime.setHeaders would do for me. Kinda hesitant forcing the users hand so much.

What others think?

Reply:
I thought it handy. It encodes url, sets Location header, sets response text or html (not sure if they're that needed tho, but koa and express for example have them). Also we can add functionality to whitelist only allowed redirect urls. Otherwise, it can be a security issue.

Reply:
yeah probably not the best place for whitelisting to be. The better place will be openapi spec in case redirect uri is give in request parameters. In case it's hardcoded/coming from config it's already should be considered allowed.